aggregation/quantile/cm/stream.go

```diff
@@ -311,7 +345,9 @@ func (s *stream) addToMinHeap(heap *minHeap, value float64) {
 	// and return the current heap to the pool
 	if len(curr) == cap(curr) {
 		newHeap := s.floatsPool.Get(2 * len(curr))
-		newHeap = append(newHeap, curr...)
+		for i := 0; i < len(curr); i++ {
```

Comment:
I don't think it allocates a new slice for `curr..` but creates a subslice for it instead. I wrote some [quick benchmarks](https://github.com/jeromefroe/golang_benchmarks#append) and the `curr..` approach was actually much faster and didn't cause any allocations.